docs/dapps/getting-started.md

```diff
@@ -8,91 +8,531 @@ import UntranslatedPageText from "@site/src/components/UntranslatedPageText";
+You can see that our contract is emitting some `event` and define an `error` code. Read the following for more info on the [events](https://wiki.alephium.org/ralph/getting-started#events) and the [error handling](https://wiki.alephium.org/ralph/getting-started#error-handling).
 
-## Compile your contract
+This is followed by five access method for the different contract's arguments.
 
-Next, change the workspace to the tutorial project:
+The latest function is where the magic happen:
 
+```rust
+@using(assetsInContract = true, updateFields = true)
+pub fn withdraw(amount: U256) -> () {
+    // Debug events can be helpful for error analysis
+    emit Debug(`The current balance is ${balance}`)
+
+    // Make sure the amount is valid
+    assert!(amount <= 2, ErrorCodes.InvalidWithdrawAmount)
+    // Functions postfixed with `!` are built-in functions.
+    transferTokenFromSelf!(callerAddress!(), selfTokenId!(), amount)
+    // Ralph does not allow underflow.
+    balance = balance - amount
+
+    // Emit the event defined earlier.
+    emit Withdraw(callerAddress!(), amount)
+}
 ```
-cd alephium-tutorial
-```
 
-and compile your contracts:
+With the `assert!` We make sure no one take more than 2 tokens at the same time.
+`transferTokenFromSelf` will actually perform the transfer of token.
+We update the `mut balance` field with the new balance, if it goes underflow and error will be raised and the transaction won't be perform.
+
+## Compile your contract
+
+The compiler needs to contact the full node in order to compile, we define the node url using with the following config file: `alephium.config.ts`
+
+```typescript
+import { Configuration } from '@alephium/cli'
+
+const configuration: Configuration<void> = {
+  defaultNetwork: 'devnet',
+  networks: {
+    devnet: {
+      //Make sure the two values match what's in your `~/.alephium/user.conf
+      nodeUrl: 'http://localhost:22973',
+      networkId: 2
+    }
+  }
+}
 
+export default configuration
 ```
+
+Make sure to use the correct host and port.
+
+Now let's compile:
+
+```sh
 npx @alephium/cli@latest compile
 ```
 
-If you take a look at `contracts/`, you should be able to find `token.ral`. The compiled artifacts are in the directory `artifacts`.
+It may ask you some confirmation to install the latest `@alephium/cli` package.
 
-This command also generates typescript code based on the compiled artifacts. The generated typescript code are in the directory `artifacts/ts`. You can interact with the alephium blockchain more conveniently by using the generated typescript code.
+You can check the produced result into `artifacts`. For example `artifacts/ts/TokenFaucet.ts` produce lots of helper functions: `at, fetchState, call*, etc`, as well as test functions.
 
 ## Test your contract
+The SDK provides unit test functionalities, which calls the contract like a normal transaction, but instead of changing the blockchain state, it returns the new contract state, transaction outputs, and events.
```

Comment:
```suggestion
The SDK provides unit test functionalities, which call the contract by sending a transaction, but instead of changing the blockchain state, it returns the new contract state, transaction outputs, and events.
```